File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/SshSetRequest.java
package org.cloudfoundry.credhub.requests;

import java.util.Objects;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.cloudfoundry.credhub.ErrorMessages;
import org.cloudfoundry.credhub.credential.SshCredentialValue;

public class SshSetRequest extends BaseCredentialSetRequest<SshCredentialValue> {

  @NotNull(message = ErrorMessages.MISSING_VALUE)
  @Valid
  @JsonProperty("value")
  private SshCredentialValue sshKeyValue;

  public SshCredentialValue getSshKeyValue() {
    return sshKeyValue;
  }

  public void setSshKeyValue(final SshCredentialValue sshKeyValue) {
    this.sshKeyValue = sshKeyValue;
  }

  @Override
  public SshCredentialValue getCredentialValue() {
    return sshKeyValue;
  }

  @Override
  public GenerationParameters getGenerationParameters() {
    return null;
  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    final SshSetRequest that = (SshSetRequest) o;
    return Objects.equals(sshKeyValue, that.sshKeyValue);
  }

  @Override
  public int hashCode() {
    return Objects.hash(sshKeyValue);
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/SshSetRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import org.cloudfoundry.credhub.ErrorMessages
import org.cloudfoundry.credhub.credential.SshCredentialValue
import java.util.Objects

class SshSetRequest : BaseCredentialSetRequest<SshCredentialValue?>() {
    @NotNull(message = ErrorMessages.MISSING_VALUE)
    @Valid
    @JsonProperty("value")
    var sshKeyValue: SshCredentialValue? = null

    override val generationParameters: GenerationParameters?
        get() = null

    override val credentialValue: SshCredentialValue?
        get() {
            return sshKeyValue
        }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as SshSetRequest?
        return sshKeyValue == that!!.sshKeyValue
    }

    override fun hashCode(): Int = Objects.hash(sshKeyValue)
}
